# MA0039 - Do not write your own certificate validation method
<!-- sources -->
Source: [DoNotUseServerCertificateValidationCallbackAnalyzer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Rules/DoNotUseServerCertificateValidationCallbackAnalyzer.cs)
<!-- sources -->

You should not write methods to validate certificates. Those methods are often used to bypass validation altogether, which is bad for security.

````csharp
System.Net.ServicePointManager.ServerCertificateValidationCallback = ... // non compliant

var handler = new System.Net.Http.HttpClientHandler();
handler.ServerCertificateCustomValidationCallback = ... // non compliant
````
