# MA0017 - Abstract types should not have public or internal constructors
<!-- sources -->
Sources: [AbstractTypesShouldNotHaveConstructorsAnalyzer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Rules/AbstractTypesShouldNotHaveConstructorsAnalyzer.cs), [AbstractTypesShouldNotHaveConstructorsFixer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer.CodeFixers/Rules/AbstractTypesShouldNotHaveConstructorsFixer.cs)
<!-- sources -->

Constructors on abstract types can be called only by derived types. Because public constructors create instances of a type, and you cannot create instances of an abstract type, an abstract type that has a public constructor is incorrectly designed.

````c#
abstract class Sample
{
    public Sample() // non-compliant
    {
    }
}

abstract class Sample
{
    protected Sample() // ok
    {
    }
}
````
