# MA0068 - Invalid parameter name for nullable attribute

````csharp
class Test
{
    [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute("unknown")] // non-compliant, "unknown" is not a parameter of the method
    public void A(string a) { }

    [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute("a")] // compliant, "a" is not a parameter of the method
    public void A(string a) { }
}
````
