# MA0062 - Non-flags enums should not be marked with `FlagsAttribute`

An enumeration marked with `FlagsAttribute` should only contain member where the value is a power of two or a combination of other values.

````csharp
[FlagsAttribute] // Noncompliant, Orange is neither a power of two, nor a combination of any of the defined values
public enum Color
{
	None    = 0,
	Red     = 1,
	Orange  = 3, 
	Yellow  = 4,
}
````