# MA0068 - Invalid parameter name for nullable attribute
<!-- sources -->
Source: [NullableAttributeUsageAnalyzer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Rules/NullableAttributeUsageAnalyzer.cs)
<!-- sources -->

````csharp
class Test
{
    [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute("unknown")] // non-compliant, "unknown" is not a parameter of the method
    public void A(string a) { }

    [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute("a")] // compliant, "a" is not a parameter of the method
    public void A(string a) { }
}
````
