# MA0039 - Do not write your own certificate validation method

You should not write your own method to validate certificates. These methods are often used to fully bypass the validation which is bad for security purpose.

````csharp
System.Net.ServicePointManager.ServerCertificateValidationCallback = ... // non compliant

var handler = new System.Net.Http.HttpClientHandler();
handler.ServerCertificateCustomValidationCallback = ... // non compliant
````