# MA0151 - DebuggerDisplay must contain valid members

Validate the content 

````c#
[DebuggerDisplay("{Display}")] // compliant
public class Dummy1
{
    public string Display { get; }
}

[DebuggerDisplay("{InvalidMember}")] // non-compliant as InvalidMember doesn't exist
public class Dummy2
{
    public string Display { get; }
}
````
