# MA0182 - Avoid uninstantiated internal classes

This analyzer detects internal classes that are never instantiated or referenced. This is similar to CA1812 but fixes several known issues:

1. **False positives with generic type arguments**: CA1812 incorrectly reports internal classes used as generic type arguments (e.g., `JsonSerializer.Deserialize<InternalClass>()`).
2. **False positives with typeof in attributes**: CA1812 incorrectly reports internal classes referenced via `typeof()` in attribute arguments.

## Non-compliant code

```csharp
internal class UnusedClass
{
    public string Name { get; set; }
}
```

## Compliant code

```csharp
// Option 1: Remove the class if it's truly unused
// (class deleted)

// Option 2: Make it static if it should only contain static members
internal static class UtilityClass
{
    public static void DoSomething() { }
}

// Option 3: Use the class
internal class UsedClass
{
    public string Name { get; set; }
}

public class Consumer
{
    public void Method()
    {
        var obj = new UsedClass();
    }
}
```

## How to fix violations

1. If the class is truly unused, remove it from the codebase.
2. If the class contains only static members, make it `static`.
3. If the class is used via generic type arguments, typeof(), or other reflection mechanisms, the analyzer correctly recognizes this and won't report a diagnostic.

## Related rules

- [CA1812: Avoid uninstantiated internal classes](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1812) - The Microsoft analyzer with known false positives
- [MA0036: Make class static](MA0036.md) - Suggests making classes with only static members static