# MA0184 - Do not use interpolated string without parameters

## Description

An interpolated string without any parameters is unnecessary and can be converted to a regular string literal for better clarity and consistency.

## Example

```csharp
// ❌ Bad: Using interpolated string without parameters
var message = $"Required attribute 'output' not found.";

// ✅ Good: Use a regular string literal
var message = "Required attribute 'output' not found.";
```

## Exceptions

This rule does not report diagnostics when:

1. The interpolated string is assigned to or converted to `FormattableString`:
   ```csharp
   FormattableString fs = $"Hello"; // OK
   ```

2. The interpolated string is used with a custom `InterpolatedStringHandler`:
   ```csharp
   void Method(CustomInterpolatedStringHandler handler) { }
   Method($"Hello"); // OK if CustomInterpolatedStringHandler has InterpolatedStringHandlerAttribute
   ```

## Configuration

This rule is **disabled by default** as it's a stylistic preference that doesn't affect runtime behavior.

To enable it, add the following to your `.editorconfig`:

```ini
dotnet_diagnostic.MA0184.severity = suggestion
```