# MA0184 - Do not use interpolated string without parameters as argument for interpolated string handler
<!-- sources -->
Sources: [DoNotUseInterpolatedStringWithoutParametersAnalyzer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Rules/DoNotUseInterpolatedStringWithoutParametersAnalyzer.cs), [DoNotUseInterpolatedStringWithoutParametersFixer.cs](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer.CodeFixers/Rules/DoNotUseInterpolatedStringWithoutParametersFixer.cs)
<!-- sources -->

When passing an interpolated string without any interpolation parameters to a method that accepts an interpolated string handler, you are creating unnecessary overhead. Instead, use a regular string literal.

````c#
// ❌ Bad: Creates unnecessary interpolated string handler
var x = string.Create(CultureInfo.InvariantCulture, $"string without parameters.");

// ✅ Good: Use the string literal directly
var x = "string without parameters.";
````

````c#
// ❌ Bad: No interpolation parameters
var x = string.Create(CultureInfo.InvariantCulture, $"literal text");

// ✅ Good: Has interpolation parameters - handler is needed
var x = string.Create(CultureInfo.InvariantCulture, $"Current time is {DateTime.Now:D}.");
````
