# MA0125 - The list of log parameter types contains an invalid type

The configuration file for the rule [MA0124](./MA0124.md) contains a type that is not found in the current compilation.

````
PropName;System.Unknown # Report diagnostic
````
