# MA0115 - Unknown component parameter

Detect usage of invalid parameter in Razor components.

```razor
@* SampleComponent *@
<p>@Text</p>

@code {
    [Parameter]
    public string Text { get; set; }
}
```

```razor
<SampleComponent
    Text="Dummy"
    InvalidParameter="Dummy" />  // Report diagnostic as InvalidParameter does not exist in SampleComponent
```
